src/providers/data_provider/dp_target_id.c

```diff
@@ -572,14 +793,41 @@ static errno_t dp_get_account_info_initgroups_step(struct tevent_req *req)
 {
     struct dp_get_account_info_state *state;
     struct tevent_req *subreq;
-    errno_t ret;
 
     state = tevent_req_data(req, struct dp_get_account_info_state);
 
     if (state->initgroups == false) {
         return EOK;
     }
 
+    /* Create subrequest to handle SR data */
+    subreq = dp_sr_resolve_groups_send(state, state->ev, state->reply,
```

Comment:
Hi,

currently you call `dp_sr_resolve_groups_send()` e.g. on all initgroup requests even if session recording is not enabled. Would it be possible to check `sc_conf.scope` or similar to check if resolving all group is needed or not?

bye,
Sumit

Reply:
Good point, I added a check for this in the beginning of dp_sr_resolve_groups_send() to shortcut return if the scope == none

Reply:
Hi,

thanks. I think you can even be more specific and add `scope == SESSION_RECORDING_SCOPE_SOME && groups != NULL` and `scope == SESSION_RECORDING_SCOPE_ALL && exclude_groups != NULL`.

bye,
Sumit

Reply:
Fixed, thank you.